src/ProjectTemplates/BlazorWasm.ProjectTemplates/content/BlazorWasm-CSharp/Shared/Authorization/UserInfo.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+
+namespace BlazorWasm_CSharp.Shared.Authorization
+{
+    public class UserInfo
+    {
+        public static readonly UserInfo Anonymous = new UserInfo();
+
+        public UserInfo() { }
+
+        public UserInfo(ClaimsPrincipal claimsPrincipal)
+        {
+            var identity = claimsPrincipal.Identity;
+            IsAuthenticated = identity.AuthenticationType != null;
+            if (identity is ClaimsIdentity claimsIdentity)
+            {
+                NameClaimType = claimsIdentity.NameClaimType;
+                RoleClaimType = claimsIdentity.RoleClaimType;
```

Comment:
Could the server and client ever have different opinions about `NameClaimType` and `RoleClaimType`?

What I really mean is, if we didn't pass this info, and the client just populated its claimsprincipal with all the claims it's recieved, wouldn't `User.Identity.Name` and the "roles" APIs still work OK? They definitely did work OK like that in my earlier prototypes, but I don't know if there are some extra edge cases you're accounting for here.

Reply:
No, they depend on the NameClaimType and RoleClaimType afaik.

Reply:
It might have worked because you were using default values for both.